simplified-viewer-audiobook/src/main/java/org/nypl/simplified/viewer/audiobook/AudioBookPlayerActivity.kt

```diff
@@ -395,6 +401,38 @@ class AudioBookPlayerActivity : AppCompatActivity(),
     }
   }
 
+  private fun loadAndConfigureExtensions(): List<PlayerExtensionType> {
+    val extensions =
+      ServiceLoader.load(PlayerExtensionType::class.java)
+        .toList()
+
+    val services = Services.serviceDirectory()
+    this.loadAndConfigureFeedbooks(services, extensions)
+    return extensions
+  }
+
+  private fun loadAndConfigureFeedbooks(
+    services: ServiceDirectoryType,
+    extensions: List<PlayerExtensionType>
+  ) {
+    val feedbooksConfigService =
+      services.optionalService(AudioBookFeedbooksServiceType::class.java)
+
+    if (feedbooksConfigService != null) {
+      this.log.debug("feedbooks configuration service is available; configuring extension")
+      val extension =
+        extensions.filter { ex -> ex is FeedbooksPlayerExtension }
```

Comment:
Pretty sure you can simplify this using the built-in [filterIsInstance](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-is-instance.html) method.

```kotlin
val extension = extensions
    .filterIsInstance<FeedbooksPlayerExtension>()
    .firstOrNull()
```

Actually, you may be able to simplify it even further like this:

```kotlin
val extension = extensions.firstOrNull { it is FeedbooksPlayerExtension }
```

Reply:
So it wasn't just my imagination. I was looking for the method that had that signature and just couldn't see it... :roll_eyes: